x-pack/plugins/security/public/plugin.ts

```diff
@@ -28,18 +43,48 @@ export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPlugi
     this.sessionTimeout = new SessionTimeout(notifications, sessionExpired, http, tenant);
     http.intercept(new SessionTimeoutHttpInterceptor(this.sessionTimeout, anonymousPaths));
 
+    this.license$ = licensing.license$;
+
     return {
       anonymousPaths,
       sessionTimeout: this.sessionTimeout,
     };
   }
 
-  public start() {
+  public start(core: CoreStart) {
     this.sessionTimeout.start();
+
+    const securityLicenseService = new SecurityLicenseService().setup();
+    this.licenseSubscription = this.license$.subscribe(rawLicense => {
+      securityLicenseService.update(rawLicense);
+      const showSecurityLinks = securityLicenseService.license.getFeatures().showLinks;
+
+      const shouldRegisterNavControl =
```

Comment:
This logic feels like it shouldn't be part of the plugin's `start` lifecycle event, and should be moved to a "security nav control service". Is there a reason we need to have this logic at the main plugin level?

Reply:
nope, we can certainly move this.